src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,137 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconCross16, IconEye, IconEyeOff } from '@foundation/Icon';
+import { useFocusRing } from '@react-aria/focus';
+import { FOCUS_STYLE } from '@utilities/focusStyle';
+import { GetStatusIcon } from '@utilities/input';
+import { merge } from '@utilities/merge';
+import { Validation, validationClassMap } from '@utilities/validation';
+import { useCallback, useRef, useState } from 'react';
+import { InputBaseProps, InputTypes } from '../../types/input';
+import { useMemoizedId } from '@hooks/useMemoizedId';
+
+export type PasswordInputProps = { obfuscated?: boolean } & Omit<
+    InputBaseProps<string>,
+    'valueSelect' | 'decorator' | 'suffix'
+>;
+
+export const PasswordInput = ({
+    autocomplete = false,
+    clearable,
+    disabled,
+    id: propId,
+    obfuscated = true,
+    placeholder,
+    readOnly,
+    required,
+    status = Validation.Default,
+    size,
+    value,
+    onChange,
+    onFocus,
+    onBlur,
+    onKeyDown,
+    'data-test-id': dataTestId = 'fondue-password-input',
+}: PasswordInputProps) => {
+    const inputElementRef = useRef<HTMLInputElement | null>(null);
+    const [isObfuscated, setIsObfuscated] = useState<boolean>(obfuscated);
+
+    const { isFocusVisible, focusProps } = useFocusRing({ within: true, isTextInput: true });
+    const { isFocusVisible: passwordButtonIsFocusVisible, focusProps: passwordButtonFocusProps } = useFocusRing();
```

Comment:
Are those 2 `isFocusVisible` const necessary? Are we not using Fondue buttons that already have it for those extra actions?

Reply:
We can, but then we run into the issue of the input ring being visible when the buttons are actually focused.